### PR TITLE
Add a CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by schmmd**
I put this information in CONTRIBUTING.md at GitHub's prompting (see https://github.com/allenai/allennlp/community).

### CODE REVIEW

**Comment by schmmd** [comment]
I figured GitHub issues would work well for enhancement requests, but we could choose another means.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+If your code fails one of these checks, you will be expected to fix your pull request before it is considered.
+
+You can run most of these tests locally, which will be faster than waiting for cloud systems to run tests.  To see
+exactly what our continuous build system runs, look at `.travis.yml`.  The two most common checks you should run
+locally are:
+
+* `pytest -v`.  If you run this in the root AllenNLP directory, this will run all of our tests.
+* `./scripts/pylint.sh`.  This will make sure your code conforms to our automatically-checkable style guidelines.
+To see our project specific linting settings, look at `.pylintrc`.
```
Comment (joelgrus) [inline CONTRIBUTING.md]: I would add `mypy` as a check, that will be the most unusual to people

**Comment by DeNeutoy** [comment]
Could you just correct this spelling mistake in this PR too: https://github.com/allenai/allennlp/commit/d991cb981a693c724fa3885c6a35977f4062999f#diff-04c6e90faac2675aa89e2176d2eec7d8R28

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* The code compiles
+* All tests pass
+* `pylint` accepts the code style (our guidelines are based on PEP8)
+* `mypy` typechecks the Python code
+* The docs can be generated successfully
+
+If your code fails one of these checks, you will be expected to fix your pull request before it is considered.
+
+You can run most of these tests locally, which will be faster than waiting for cloud systems to run tests.  To see
+exactly what our continuous build system runs, look at `.travis.yml`.  The two most common checks you should run
```
Comment (matt-gardner) [inline CONTRIBUTING.md]: `s/two/three/`, or just remove the modifier.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+Open [a new GitHub pull request](https://github.com/allenai/allennlp/pulls) with the fix.  Make sure you have a clear
+description of the problem and the solution, and include a link to relevant issues.
+
+Once your pull request is created, our continuous build system will check your pull request.  Continuous
+build will test that:
+
+* The code compiles
+* All tests pass
+* `pylint` accepts the code style (our guidelines are based on PEP8)
+* `mypy` typechecks the Python code
```
Comment (schmmd) [inline CONTRIBUTING.md]: @matt-gardner I'd like to add something here about our code coverage checks.  Where can I find the rules those follow presently?

Comment (matt-gardner) [inline CONTRIBUTING.md]: They'd be on the page where you enabled codecov integration (the one that I can't access), I think.  Neither coverage check is required, however.

I'd probably just say something about how we strive for >90% code coverage, and your PR is a whole lot more likely to get accepted if it has proper tests.

**Pull request merged**